src/peer-connection.ts

```diff
@@ -24,14 +24,22 @@ type IceGatheringStateChangeEvent = {
   target: EventTarget | null;
 };
 
+type SdpChangeEvent = {
+  type: 'offer' | 'answer';
+  munged: boolean;
+  sdp?: string;
+};
```

Comment:
A couple of things here.

1. I think I prefer having separate events for each `createOffer`, `setLocalDescription`, and `setRemoteDescription`. It may not always be the case that the user will want to listen to all of these events, so it would be better for them to pick and choose which ones to listen to.
2. Remember that "munge" refers to modifying the SDP. But actually, we don't know if the SDP when calling `setLocalDescription` or `setRemoteDescription` will be modified. For example, someone might call `createOffer` and then `setLocalDescription` right away without munging the SDP.